icepyx/quest/dataset_scripts/argo.py

```diff
@@ -0,0 +1,492 @@
+            This kwarg is used to replace the existing pressure range in `self.presRange`.
+            Do not submit this kwarg if you would like to use the existing `self.presRange` values.
+            Input as a "shallow-limit,deep-limit" string.
+        keep_existing: boolean, default True
+            Provides the option to clear any existing downloaded data before downloading more.
+
+        Returns
+        -------
+        pd.DataFrame: DataFrame of requested data
+        """
+
+        # TODO: do some basic testing of this block and how the dataframe merging actually behaves
+        if keep_existing == False:
+            print(
+                "Your previously stored data in reg.argodata",
+                "will be deleted before new data is downloaded.",
+            )
+            self.argodata = None
+        elif keep_existing == True and hasattr(self, "argodata"):
+            print(
+                "The data requested by running this line of code\n",
+                "will be added to previously downloaded data.",
+            )
+
+        # if download is called with replaced parameters or presRange
+        if not params is None:
+            self.params = params
+
+        if not presRange is None:
+            self.presRange = presRange
+
+        # Add qc data for each of the parameters requested
+        if self.params == ["all"]:
+            pass
+        else:
+            for p in self.params:
+                if p.endswith("_argoqc") or (p + "_argoqc" in self.params):
+                    pass
+                else:
+                    self.params.append(p + "_argoqc")
+
+        # intentionally resubmit search to reset prof_ids, in case the user requested different parameters
+        self.search_data()
+
+        # create a dataframe for each profile and merge it with the rest of the profiles from this set of parameters being downloaded
+        merged_df = pd.DataFrame(columns=["profile_id"])
+        for i in self.prof_ids:
+            print("processing profile", i)
+            try:
+                profile_data = self._download_profile(i)
+                profile_df = self._parse_into_df(profile_data[0])
+                merged_df = pd.concat([merged_df, profile_df], sort=False)
+            except:
+                print('\tError processing profile {0}. Skipping.'.format(i))
+
+        # now that we have a df from this round of downloads, we can add it to any existing dataframe
+        # note that if a given column has previously been added, update needs to be used to replace nans (merge will not replace the nan values)
+        if not self.argodata is None:
+            self.argodata = self.argodata.merge(merged_df, how="outer")
+        else:
+            self.argodata = merged_df
+
+        self.argodata.reset_index(inplace=True, drop=True)
+
+        if savename:
+            self.argodata.to_csv(savename + '_argo.csv')
```

Comment:
I really like this idea, but I have some concerns about what could happen for users if they're not aware of this happening for them (e.g. overwriting existing data, assuming that's the default behavior if "write","append" etc. isn't passed in to `to_csv`; creating a whole bunch of files (potentially with intended path names, since that's what QUEST is passing in) by default). At a minimum we need to add some documentation for it and make sure we update the tutorial, which currently specifies that the data is not stored. I would also like to make it optional for the user, which may be tricky since a path is required for IS2 data (so we can't use the `path=""` default).

Reply:
Follow up: implemented this as an independent `save` function.